### PR TITLE
Ml-workspace image size reduction

### LINKED ISSUE #26: ml-workspace: image size reduction
Presently, changing ml-workspace from root to joyvan in the Dockerfile on a new layer blows up the image size (~8gb to ~14gb for cpu, ~17gb to ~30gb for gpu) due to file duplication caused by chmod and chown commands (the originals are kept on the preceding layers). I am considering two avenues to tackle this; there may be others:

**1. Rewrite the original ml-workspaces Dockerfile**. The main issue there is that I cannot get that to build: even when completely unchanged, it fails on step 50/97 while building either jupyterlab or one of its extensions, and I am having trouble even getting a log for that failure. I could try running the preceding step and building jupyterlab from the shell...

**2. Change permissions outside the build process**: Perhaps through an exec script.

Unsuccessful approaches included squashing and the devicemapper storage driver (although it may still be useful within a single layer).

### PR DESCRIPTION
**Pull request by frazs**
Initial minimization of the ml-workspace image, now working with the raw Dockerfile instead of an extension.

The following custom images are available for testing:
k8scc01covidacr.azurecr.io/ml-workspace-base:test
k8scc01covidacr.azurecr.io/ml-workspace-rstudio:test
k8scc01covidacr.azurecr.io/ml-workspace-geomatics:test

After discussing with Blair, I have split it into three images: 4.4 GB base, 7 GB rstudio (extends base with r-runtime and RStudio), and 9.6 GB geomatics (extends the rstudio image with qgis and the libraries from the geomatics notebook).

In comparison, the original (root only) ml-workspace base image is around 8 GB, and the extended non-root version with RStudio came in at 16.7.

Notable large removed components from the original image include jupyerlab (redundant to our notebook images), tensorflow and pytorch (could re-introduce in a GPU flavor if desired), JVM (could re-introduce in a Scala flavour if desired), and chromium (went with firefox-only, pending a reason otherwise). 

Resolves #26, specifically by eliminating the chown/chmod bloat from image extension. Further minimizations may still be possible, but in the meantime this seems to be an appropriate state for testing and feedback.

The Dockerfile could still use cleanup and perhaps some refactoring/standardisation (e.g. some things are installed directly in the Dockerfile while others through refrenced scripts). I have kept this version close to the original, with most things commented out instead of removed or re-ordered, in order to easily revert any undiscovered breaking or undesired changes. Once there has been some testing/feedback I can clean it up in a future PR.

### CODE REVIEW

**Comment by frazs** [comment]
Per initial discussions MLK and XGBoost had also been removed from the base. Let me know if either or both should be back for current or future projects, and if they're applicable across the board or more appropriate as part of a new separate image.

File: ml-workspace/base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,1150 @@
...
+    "org.label-schema.name"="Machine Learning Workspace" \
+    "org.label-schema.description"="All-in-one web-based development environment for machine learning." \
+    "org.label-schema.usage"="https://github.com/ml-tooling/ml-workspace" \
+    "org.label-schema.url"="https://github.com/ml-tooling/ml-workspace" \
+    "org.label-schema.vcs-url"="https://github.com/ml-tooling/ml-workspace" \
+    "org.label-schema.vendor"="ML Tooling" \
+    "org.label-schema.version"=$WORKSPACE_VERSION \
+    "org.label-schema.schema-version"="1.0" \
+    "org.label-schema.vcs-ref"=$ARG_VCS_REF \
+    "org.label-schema.build-date"=$ARG_BUILD_DATE
```
[the remark this answers is not in the document]
Comment (frazs) [reply ml-workspace/base/Dockerfile:1091]: How about just removing this? At least for now, pending any standardized system of LABEL'ing our other images.

**Comment by brendangadd** [comment]
Changes commited into new repository: StatCan/kubeflow-desktop
